repo2docker/utils.py

```diff
@@ -99,3 +100,166 @@ def validate(self, obj, value):
             raise TraitError('{val} is not a valid memory specification. Must be an int or a string with suffix K, M, G, T'.format(val=value))
         else:
             return int(float(num) * self.UNIT_SUFFIXES[suffix])
+
+
+class ImageNameValidator:
+    """
+    Given a docker image_name, check if the image_name conforms to the restrictions placed by docker.
+
+    Class defines the regex patterns based off of the definitions in
+    https://github.com/docker/distribution/blob/master/reference/regexp.go. There are some modifications as noted below.
+    """
+
+    def __init__(self):
```

Comment:
This is great, but I think this should just be a function rather than a class. I'd consider using the `re.VERBOSE` option (https://docs.python.org/3/library/re.html#re.VERBOSE) to make this into a single multi-line regex, and just make a 'is_valid_imagename' function here that can be used. I also recommend watching (https://www.youtube.com/watch?v=o9pEzgHorH0) for a good sense of when a class is useful and when it isn't in Python.

Reply:
Changed. Makes more sense. Thanks for the link to the video.